### PR TITLE
Issue 15

### PR DESCRIPTION
**Pull request by chibenwa**
This pull request provides : 

 - Signature stuff
 - Continuation token validation and generation stuff
 - Modification of the servlet in order to handle Continuation tokens in the right way.

### CODE REVIEW

File: protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationServlet.java [hunk 1]
```diff
@@ -66,7 +70,10 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
 		} catch (BadRequestException e) {
 			LOG.warn("Invalid authentication request received.", e);
 			resp.sendError(HttpServletResponse.SC_BAD_REQUEST);
-		}
+		} catch (InternalErrorException e) {
```
Comment (mbaechler) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationServlet.java]: indent is broken

Comment (chibenwa) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationServlet.java]: Sorry, I don't know what I have done with my IDE... 
I corrected my indentation parameters and updated the given commit.

Comment (chibenwa) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationServlet.java]: @rouazana : Don't you indent with TAB ????

File: protocols/jmap/src/main/java/org/apache/james/jmap/model/ContinuationToken.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        this.expirationDate = expirationDate;
+        this.signature = signature;
+    }
+
+    public static ContinuationToken fromString(String tokenString) throws MalformedContinuationTokenException {
+        Preconditions.checkArgument(! Strings.isNullOrEmpty(tokenString), "Serialized continuation token should not be null or empty");
+        if (StringUtils.countMatches(tokenString, SEPARATOR) < 2) {
+            throw new MalformedContinuationTokenException("Token " + tokenString + " does not have enough parts");
+        }
+        String signature = extractLastTokenComponent(tokenString);
```
Comment (mbaechler) [inline protocols/jmap/src/main/java/org/apache/james/jmap/model/ContinuationToken.java]: sorry but this code is still not ok.
* `tokenString` is not a good name.
* `ContinuationToken(extractRemainingPayload(payload)` doesn't say (at all) what this first arg is about
* `substring` and `lastIndexOf` is really a thing of the past

I suggest you :
1. split the string on separator
2. check the list size
3. extract every parts of the list into a meaningful variable
4. do things with these variables

I understand you need to iterate from the end to allow the token to be present in the first part, but you can just do a reverse iteration on the split result.

**Pull request merged**